### PR TITLE
 Make RobustConnection more robust

### PR DESCRIPTION
**Pull request by decaz**
* Added `RuntimeError` to `CONNECTION_EXCEPTIONS`
* Handle `CONNECTION_EXCEPTIONS` at channel reconnect

### CODE REVIEW

File: aio_pika/connection.py [hunk 1]
```diff
@@ -52,12 +52,12 @@ def __init__(self, url, loop=None, **kwargs):
         self.kwargs = self._parse_kwargs(kwargs or self.url.query)
 
         self.connection = None     # type: aiormq.Connection
-        self.close_callbacks = set()
+        self._close_callbacks = set()
```
Comment (mosquito) [inline aio_pika/connection.py]: why you want to hide this? User may use it directly It's legal.

Comment (decaz) [reply aio_pika/connection.py]: It's the only place where callbacks set is public, - all other callback sets are private. I've made this to uniform callbacks storages.

Comment (mosquito) [reply aio_pika/connection.py]: 🤔

Comment (decaz) [reply aio_pika/connection.py]: @mosquito what do you think?

Comment (mosquito) [reply aio_pika/connection.py]: Uniform callback storage sounds good. Let's do it.

**Comment by mosquito** [comment]
@decaz my vision of uniform callback api requested in #239

**Comment by decaz** [comment]
@mosquito I've removed all changes of callback usage in favor of #239.

**Pull request merged**